### PR TITLE
make swagger deterministic

### PR DESCRIPTION
**Pull request by decentralgabe**
as per https://github.com/TBD54566975/ssi-service/issues/447 swagger generation is non deterministic

this change is a stop-gap to make swagger deterministic until swaggo/swag is fixed

added back a ci check to fail if swagger is out of date

### CODE REVIEW

File: doc/overrides.swaggo [hunk 1]
```diff
@@ -0,0 +1,2 @@
+replace    cryptosuite.LDKeyType string
+replace    cryptosuite.SignatureType string
```
Comment (decentralgabe) [inline doc/overrides.swaggo:2]: exclude the two problematic types for now

**Comment by michaelneale** [comment]
nice one

File: magefile.go [hunk 2]
```diff
@@ -105,25 +105,24 @@ func Integration() error {
...
+		return err
 	}
+
 	// One of the dependencies we have (antlr4) does not play nicely with swaggo, but we need to enable dependencies
 	// because many of our external API objects have ssi-sdk objects. We can work around this by setting depth. You can
 	// see a discussion of this topic in https://github.com/swaggo/swag/issues/948.
 	// We also set parseGoList because it's the only way parseDepth works until the following is fixed:
 	// https://github.com/swaggo/swag/issues/1269
-	return sh.Run(swagCommand, "init", "-g", "cmd/ssiservice/main.go", "--pd", "-o", "doc", "-ot", "go,yaml", "--parseInternal")
+	return sh.Run(swagCommand, "init", "-g", "cmd/ssiservice/main.go", "--overridesFile", "doc/overrides.swaggo", "--pd", "--parseInternal", "-o", "doc", "-ot", "go,yaml")
```
Comment (andresuribe87) [inline magefile.go:125]: If the comment above doesn't apply anymore can you update please?

Comment (decentralgabe) [reply magefile.go:121]: done

Comment (decentralgabe) [reply magefile.go:121]: done

**Review by andresuribe87 — APPROVED** [review]

**Pull request merged**